### PR TITLE
FE-335 Use a hash of the apiKey as the named-index for glympse-adapter's credentials caching

### PR DESCRIPTION
**Pull request by Spiva**
@Glympse/web PTAL

### CODE REVIEW

**Comment by adubovsky** [comment]
LGTM. **Please note** that old auth data will be ignored after merging this (e.g. user will start with new account flow in the frapp).

**Comment by j5kay** [comment]
@adubovsky - yes, has been my hesitancy on integrating. To prevent a flood of new accounts from this change, need to look for the old accounts, move to the new key hash, then delete the old account. Pretty sure this affects a PR on the viewer as well.

**Comment by adubovsky** [comment]
@Glympse/web, PTAL.

File: app/src/lib/utils.js [hunk 1]
```diff
@@ -175,6 +175,19 @@ define(function(require, exports, module)
...
+		{
+			var hash = 0, i, chr, len;
+			if (str.length === 0) return hash;
+			for (i = 0, len = str.length; i < len; i++)
+			{
+				chr = str.charCodeAt(i);
+				hash = ((hash << 5) - hash) + chr;
+				hash |= 0; // Convert to 32bit integer
+			}
+			return hash;
```
Comment (j5kay) [inline app/src/lib/utils.js]: I know we want to use a simpler hash algorithm, but munging into an int is a little too lean. Would like to expand the namespace to use at least an alpha-numeric representation. Can you find/review/implement? (will need to also add to the adapter version)

**Comment by adubovsky** [comment]
Changed hash algorithm to md5. @j5kay, PTAL.

**Comment by j5kay** [comment]
LGTM

**Pull request merged**